### PR TITLE
migration(crons): Remove invalidly slugged monitors

### PR DESCRIPTION
**Pull request by rjo100**
Correctly marks monitors as `PENDING_DELETION` instead of trying to delete them mid run.

Fixes https://github.com/getsentry/sentry/pull/52471

### CODE REVIEW

File: src/sentry/migrations/0514_slugify_invalid_monitors.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                        ObjectStatus.DELETION_IN_PROGRESS,
+                    ]
+                )
+                .first()
+            )
+            if rule:
+                rule.status = ObjectStatus.PENDING_DELETION
+                rule.save()
+
+        delete_monitor.status = ObjectStatus.PENDING_DELETION
```
Comment (rjo100) [inline src/sentry/migrations/0514_slugify_invalid_monitors.py:52]: will this eventually be cleaned up? I can't seem to call `ScheduledDeletion.schedule()` directly in here as it's a class method

Comment (wedamija) [reply src/sentry/migrations/0514_slugify_invalid_monitors.py:52]: If you're using scheduled deletions you'll need to create the rows. @ceorourke  did this recently for `Rule` deletions https://github.com/getsentry/sentry/blob/8d4b9ccb38443ebd9d209c7f40b38f1917a191be/src/sentry/migrations/0507_delete_pending_deletion_rules.py#L19-L43

Comment (markstory) [reply src/sentry/migrations/0514_slugify_invalid_monitors.py:52]: Regionless is fine for now.

File: src/sentry/migrations/0515_slugify_invalid_monitors.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        },
+    )
+
+
+def delete_rules(apps, schema_editor):
+    Rule = apps.get_model("sentry", "Rule")
+    RegionScheduledDeletion = apps.get_model("sentry", "RegionScheduledDeletion")
+    for rule in RangeQuerySetWrapperWithProgressBar(Rule.objects.all()):
+        if rule.status in (ObjectStatus.PENDING_DELETION, ObjectStatus.DISABLED):
+            schedule(RegionScheduledDeletion, rule, days=0)
```
Comment (wedamija) [inline src/sentry/migrations/0515_slugify_invalid_monitors.py:34]: Looks unused?

File: src/sentry/migrations/0515_slugify_invalid_monitors.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        # Nothing to migrate if the monitor already has a valid slug
+        if monitor_slug == slugified:
+            continue
+
+        try:
+            monitor.slug = slugified
+            monitor.save()
+        except IntegrityError:
+            # If there is a collision, delete the old monitor as the new one is receiving all check-ins
+            monitors_to_clean_up.append(monitor.id)
```
Comment (wedamija) [inline src/sentry/migrations/0515_slugify_invalid_monitors.py:57]: Any reason we don't just perform the cleanup here? If there are a lot of these monitors we might bring them all into memory in the later query

File: src/sentry/migrations/0515_slugify_invalid_monitors.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                    ]
+                )
+                .first()
+            )
+            if rule:
+                rule.status = ObjectStatus.PENDING_DELETION
+                rule.save()
+
+        delete_monitor.status = ObjectStatus.PENDING_DELETION
+        delete_monitor.save()
```
Comment (wedamija) [inline src/sentry/migrations/0515_slugify_invalid_monitors.py:80]: You still need to call schedule here

File: src/sentry/migrations/0515_slugify_invalid_monitors.py (hunk 3 above)
Comment (rjo100) [reply src/sentry/migrations/0515_slugify_invalid_monitors.py:51]: ah good point. can do that

Comment (rjo100) [reply src/sentry/migrations/0515_slugify_invalid_monitors.py:51]: just got lazy about rewrites

File: src/sentry/migrations/0515_slugify_invalid_monitors.py (hunk 4 above)
Comment (rjo100) [reply src/sentry/migrations/0515_slugify_invalid_monitors.py]: yeah had a bunch of local changes I didn't realize got overwritten with a bad merge/update migration script

File: src/sentry/migrations/0515_slugify_invalid_monitors.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+def migrate_monitor_slugs(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+    Rule = apps.get_model("sentry", "Rule")
+    RegionScheduledDeletion = apps.get_model("sentry", "RegionScheduledDeletion")
+    ScheduledDeletion = apps.get_model("sentry", "ScheduledDeletion")
+
+    MAX_SLUG_LENGTH = 50
+    monitors_to_clean_up = []
```
Comment (wedamija) [inline src/sentry/migrations/0515_slugify_invalid_monitors.py:36]: Nit: Unused

**Review by wedamija — APPROVED** [review]

**Pull request merged**